sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -870,7 +874,7 @@ static int sof_card_dai_links_create(struct device *dev,
 	 * link ID used by sdw link. The dai_id is not always equal
 	 * to ID for DAI link ID may be changed by some sdw codecs.
 	 */
-	dai_id = links[id - 1].id + 1;
```

Comment:
@RanderWang for my education, I understand the comment above about the DAI link ID being modified by codecs. Where is this being done?

Reply:
````
static void rt715_init(const struct snd_soc_acpi_link_adr *link,
		       struct snd_soc_dai_link *dai_links,
		       struct codec_info *info,
		       bool playback)
{
	/*
	 * DAI ID is fixed at SDW_DMIC_DAI_ID for 715 to
	 * keep sdw DMIC and HDMI setting static in UCM
	 */
	if (sof_rt711_rt1308_rt715_quirk & SOF_RT715_DAI_ID_FIX)
		dai_links->id = SDW_DMIC_DAI_ID;
}
````

Reply:
Thanks @RanderWang. Can we please amend the comment above to say that it is changed in the codec init function for others who may have the same question?

Reply:
yes, I will update the comments